### PR TITLE
feat: Make appDir default as stable since Next.js 13

### PR DESCRIPTION
**Pull request by jrobsontull**
Since the new app router was made stable in Next.js 13, we should consider making this the default way to initialise `next-auth` within projects. I had to spend some time learning how to integrate the `/app` directory with `next-auth` and I think a full working example repository would be really helpful.

It is probably worth maintaining the current `/pages` example repository since support for this will not be going away anytime soon and most codebases likely make use of this version of the Next.js router.

I know that there is an existing [next-13-auth-app](https://github.com/balazsorban44/nextjs-13-next-auth-app) repository but I could not find a link to this in the documentation. I still think it would be good to maintain a similar repository as the current `next-auth-example` to make it easier for users to migrate and for new users to understand how to integrate `next-auth` fully with Next.js 13. Perhaps we need an official `next-auth-next-13-example`?

### CODE REVIEW

**Comment by DougInAMug** [comment]
Thanks for bring that repo to my attention @jrobsontull ! Whether or not it's default, would be very useful to have more prominently linked in main docs.

**Comment by bourgeois247** [comment]
Thank you @jrobsontull, your PR saved me from the nightmare trying to figure out what I was doing wrong. I had an inkling that it had to do with `route.ts` somehow, but you PR clarified exactly how to make the changes.

**Comment by pedroapfilho** [comment]
This PR deserves its own repo IMO, something like `next-auth-appdir`

**Comment by jrobsontull** [comment]
> This PR deserves its own repo IMO, something like `next-auth-appdir`

@pedroapfilho Given the fact that people are still confused in the [original PR](https://github.com/nextauthjs/next-auth/pull/6777) and [issue/discussio](https://github.com/nextauthjs/next-auth/issues/6792)n for route handler support, I think this is probably the best approach.

@balazsorban44 any thoughts on this?

**Comment by philip-ulrich** [comment]
Also wanted to pop in and say thanks for this. Should probably just get merged. 😆

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "Lluis Agusti <hi@llu.lu>"
   ],
   "dependencies": {
-    "next": "latest",
+    "next": "^13.4.5",
```
Comment (ImBIOS) [inline package.json:21]: I recommend to using 'latest' for project with context as example, and let the exact version saved in the lockfile.

**Review by ImBIOS — CHANGES_REQUESTED** [review]

File: lib/auth/AuthProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+// Use of the <SessionProvider> is mandatory to allow components that call
+// `useSession()` anywhere in your application to access the `session` object
+// We use client hydration to establish the <SessionProvider />
+export default function AuthProvider({
+  children,
+}: {
+  children?: React.ReactNode;
+}) {
+  return <SessionProvider>{children}</SessionProvider>;
```
Comment (ImBIOS) [inline lib/auth/AuthProvider.tsx:13]: Where is the session passed to SessionProvider?

**Comment by balazsorban44** [comment]
Closing as per the README https://github.com/nextauthjs/next-auth-example#readme

**Pull request closed without merging**